cmds/basename/basename_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/testutil"
+)
+
+type test struct {
+	flags      []string
+	out        string
+	stdErr     string
+	exitStatus int
+}
+
+func TestBasename(t *testing.T) {
+
+	var tests = []test{
+		{
+			flags:      []string{"foo.h", ".h"},
+			out:        "foo\n",
+			stdErr:     "",
+			exitStatus: 0,
+		},
+		{
+			flags:      []string{"/bar/baz/biz/foo.h", "bar/baz/biz"},
+			out:        "foo.h\n",
+			stdErr:     "",
+			exitStatus: 0,
+		},
+		{
+			// Found this behavior kind of odd, if you provide just the suffix
```

Comment:
The existing behaviour makes sense. It is part of step 6 in the spec: http://pubs.opengroup.org/onlinepubs/009695299/utilities/basename.html

> 6. If the suffix operand is present, _is not identical to the characters remaining in string_, and is identical to a suffix of the characters remaining in string, the suffix suffix shall be removed from string. Otherwise, string is not modified by this step. It shall not be considered an error if suffix is not found in string.